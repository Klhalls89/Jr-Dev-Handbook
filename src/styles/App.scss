* {
  box-sizing: border-box;
}

body {
/*  background-image: url(pinkpaper.jpg);
  object-fit: cover;*/
  background-color: mediumvioletred;
}

.logo {
  padding-top: 20px;
  font-family: $logoFont;
  font-size: 60px;
  text-align: center;
}
