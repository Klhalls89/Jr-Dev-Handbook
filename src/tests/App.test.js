import React from "react";
import ReactDOM from "react-dom";
import App from "../scripts/App";
import { shallow } from 'enzyme';

let mockData = [
    { question: 'What does html mean',
      answers: [ 'hypertext markup language',
                  'false',
                  'incorrect',
                  'untrue'
                ]
    },
    { question: 'What does css mean',
      answers: [ 'cascading sytle sheet',
                  'false',
                  'incorrect',
                  'untrue'
                ]
    }
]

describe('App testing', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<App />);
    wrapper.setState({
      questions: mockData,
      results: ''
    })
  })

  it("should match the snapshot with mock state", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('valid answer', () => {

    it('should shift the first question out of question array', () => {
      expect(wrapper.state('questions')).toHaveLength(2)
      wrapper.instance().validAnswer()
      expect(wrapper.state('questions')).toHaveLength(1)
    })

    it('should update results in state', () => {
      expect(wrapper.state('results')).toBe('')
      wrapper.instance().validAnswer()
      expect(wrapper.state('results')).toBe('You got this')
    })

  })

  describe('invalid answer', () => {
    const shiftedQuestion = { question: 'What does html mean',
    answers: [  'hypertext markup language',
    'false',
    'incorrect',
    'untrue'
      ]
    }

    it('should shift the first question out of question array then push it to the back', () => {
      expect(wrapper.state('questions')[0]).toEqual(shiftedQuestion)
      wrapper.instance().invalidAnswer()
      expect(wrapper.state('questions')[1]).toEqual(shiftedQuestion)
    })

    it('should update results in state', () => {
      expect(wrapper.state('results')).toBe('')
      wrapper.instance().invalidAnswer()
      expect(wrapper.state('results')).toBe('You can do better')
    })

  })

})