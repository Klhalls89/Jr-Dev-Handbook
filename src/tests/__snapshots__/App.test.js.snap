// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing should match the snapshot with mock state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="header-sect"
        >
          <h1
            className="logo"
          >
            The Jr. Dev Handbook
          </h1>
        </section>,
        <section
          className="instructions-sect"
        >
          <h2>
            Instructions:
          </h2>
          <p>
            Welcome to the junior developer's handbook. It's a series of flash cards with terms and ideas you should know. Anything you get right will be taken out of your flashcard deck. Anything you get wrong will be asked again. Start studying!
          </p>
        </section>,
        <section
          className="card-sect"
        >
          <Card
            currentQuestion={
              Object {
                "answers": Array [
                  "hypertext markup language",
                  "false",
                  "incorrect",
                  "untrue",
                ],
                "question": "What does html mean",
              }
            }
            invalidAnswer={[Function]}
            validAnswer={[Function]}
          />
        </section>,
        <section
          className="results-sect"
        >
          <h2>
            Results:
          </h2>
          <p
            className="results"
          >
            
          </p>
          <a
            href="http://frontend.turing.io/lessons/"
          >
            Learn more about becoming a junior developer
          </a>
        </section>,
      ],
      "className": "quiz-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="logo"
          >
            The Jr. Dev Handbook
          </h1>,
          "className": "header-sect",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Jr. Dev Handbook",
            "className": "logo",
          },
          "ref": null,
          "rendered": "The Jr. Dev Handbook",
          "type": "h1",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Instructions:
            </h2>,
            <p>
              Welcome to the junior developer's handbook. It's a series of flash cards with terms and ideas you should know. Anything you get right will be taken out of your flashcard deck. Anything you get wrong will be asked again. Start studying!
            </p>,
          ],
          "className": "instructions-sect",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Instructions:",
            },
            "ref": null,
            "rendered": "Instructions:",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to the junior developer's handbook. It's a series of flash cards with terms and ideas you should know. Anything you get right will be taken out of your flashcard deck. Anything you get wrong will be asked again. Start studying!",
            },
            "ref": null,
            "rendered": "Welcome to the junior developer's handbook. It's a series of flash cards with terms and ideas you should know. Anything you get right will be taken out of your flashcard deck. Anything you get wrong will be asked again. Start studying!",
            "type": "p",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Card
            currentQuestion={
              Object {
                "answers": Array [
                  "hypertext markup language",
                  "false",
                  "incorrect",
                  "untrue",
                ],
                "question": "What does html mean",
              }
            }
            invalidAnswer={[Function]}
            validAnswer={[Function]}
          />,
          "className": "card-sect",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currentQuestion": Object {
              "answers": Array [
                "hypertext markup language",
                "false",
                "incorrect",
                "untrue",
              ],
              "question": "What does html mean",
            },
            "invalidAnswer": [Function],
            "validAnswer": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Results:
            </h2>,
            <p
              className="results"
            >
              
            </p>,
            <a
              href="http://frontend.turing.io/lessons/"
            >
              Learn more about becoming a junior developer
            </a>,
          ],
          "className": "results-sect",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Results:",
            },
            "ref": null,
            "rendered": "Results:",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "results",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Learn more about becoming a junior developer",
              "href": "http://frontend.turing.io/lessons/",
            },
            "ref": null,
            "rendered": "Learn more about becoming a junior developer",
            "type": "a",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="header-sect"
          >
            <h1
              className="logo"
            >
              The Jr. Dev Handbook
            </h1>
          </section>,
          <section
            className="instructions-sect"
          >
            <h2>
              Instructions:
            </h2>
            <p>
              Welcome to the junior developer's handbook. It's a series of flash cards with terms and ideas you should know. Anything you get right will be taken out of your flashcard deck. Anything you get wrong will be asked again. Start studying!
            </p>
          </section>,
          <section
            className="card-sect"
          >
            <Card
              currentQuestion={
                Object {
                  "answers": Array [
                    "hypertext markup language",
                    "false",
                    "incorrect",
                    "untrue",
                  ],
                  "question": "What does html mean",
                }
              }
              invalidAnswer={[Function]}
              validAnswer={[Function]}
            />
          </section>,
          <section
            className="results-sect"
          >
            <h2>
              Results:
            </h2>
            <p
              className="results"
            >
              
            </p>
            <a
              href="http://frontend.turing.io/lessons/"
            >
              Learn more about becoming a junior developer
            </a>
          </section>,
        ],
        "className": "quiz-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="logo"
            >
              The Jr. Dev Handbook
            </h1>,
            "className": "header-sect",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The Jr. Dev Handbook",
              "className": "logo",
            },
            "ref": null,
            "rendered": "The Jr. Dev Handbook",
            "type": "h1",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Instructions:
              </h2>,
              <p>
                Welcome to the junior developer's handbook. It's a series of flash cards with terms and ideas you should know. Anything you get right will be taken out of your flashcard deck. Anything you get wrong will be asked again. Start studying!
              </p>,
            ],
            "className": "instructions-sect",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Instructions:",
              },
              "ref": null,
              "rendered": "Instructions:",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to the junior developer's handbook. It's a series of flash cards with terms and ideas you should know. Anything you get right will be taken out of your flashcard deck. Anything you get wrong will be asked again. Start studying!",
              },
              "ref": null,
              "rendered": "Welcome to the junior developer's handbook. It's a series of flash cards with terms and ideas you should know. Anything you get right will be taken out of your flashcard deck. Anything you get wrong will be asked again. Start studying!",
              "type": "p",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Card
              currentQuestion={
                Object {
                  "answers": Array [
                    "hypertext markup language",
                    "false",
                    "incorrect",
                    "untrue",
                  ],
                  "question": "What does html mean",
                }
              }
              invalidAnswer={[Function]}
              validAnswer={[Function]}
            />,
            "className": "card-sect",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "currentQuestion": Object {
                "answers": Array [
                  "hypertext markup language",
                  "false",
                  "incorrect",
                  "untrue",
                ],
                "question": "What does html mean",
              },
              "invalidAnswer": [Function],
              "validAnswer": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Results:
              </h2>,
              <p
                className="results"
              >
                
              </p>,
              <a
                href="http://frontend.turing.io/lessons/"
              >
                Learn more about becoming a junior developer
              </a>,
            ],
            "className": "results-sect",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Results:",
              },
              "ref": null,
              "rendered": "Results:",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "results",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Learn more about becoming a junior developer",
                "href": "http://frontend.turing.io/lessons/",
              },
              "ref": null,
              "rendered": "Learn more about becoming a junior developer",
              "type": "a",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
